TWBMSWSensor.cpp

```diff
@@ -31,41 +30,44 @@ bool TWBMSWSensor::OpenPort(size_t speed, uint32_t config, uint8_t rx, uint8_t t
     return ModBusRtuClass::begin(speed, config, rx, tx);
 }
 
+void TWBMSWSensor::SetModbusAddress(uint8_t address)
+{
+    this->Address = address;
+}
+
 bool TWBMSWSensor::GetFwVersion(uint32_t* version)
 {
-    uint32_t out;
-    uint32_t number;
-    uint16_t fw_v[0x10];
-    size_t i;
-    size_t count;
+    uint16_t fw_v[16];
     uint16_t letter;
 
-    if (ModBusRtuClass::readInputRegisters(this->Address,
-                                           WBMSW_REG_FW_VERSION,
-                                           (sizeof(fw_v) / sizeof(fw_v[0x0])),
-                                           &fw_v[0x0]) == false)
+    if (!ModBusRtuClass::readInputRegisters(this->Address,
+                                            WBMSW_REG_FW_VERSION,
+                                            (sizeof(fw_v) / sizeof(fw_v[0])),
+                                            &fw_v[0]))
     {
         return (false);
     }
-    out = 0x0;
-    number = 00;
-    i = 0x0;
-    count = 0x0;
-    while ((letter = fw_v[i]) != 0x0) {
+
+    uint32_t out = 0;
+    uint32_t number = 0;
+    size_t i = 0;
+    size_t count = 0;
+    while (letter = fw_v[i]) {
         if (letter == '.') {
-            out = (out << 0x8) | number;
-            number = 0x0;
-            count = 0x0;
+            out = (out << 8) | number;
+            number = 0;
+            count = 0;
         } else {
-            number = number * 0xA + (letter - 0x30);
+            number = number * 10 + (letter - '0');
             count++;
         }
         i++;
     }
-    if (count != 0x0)
-        out = (out << 0x8) | number;
-    version[0x0] = out;
-    return (true);
+    if (count) {
+        out = (out << 8) | number;
+    }
+    version[0] = out;
```

Comment:
```suggestion
    *version = out;
```
А то кажется, что тут массив

Reply:
поправлено